sdk/component/internal/storer/local/sqlite/sqlite.go

```diff
@@ -3,42 +3,69 @@ package sqlite
 import (
 	"context"
 	"database/sql"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"path"
+	"time"
 
+	"github.com/jonboulle/clockwork"
 	_ "github.com/mattn/go-sqlite3"
+	"google.golang.org/protobuf/encoding/protojson"
 
+	"github.com/smithy-security/smithy/sdk/component/internal/storer"
+	"github.com/smithy-security/smithy/sdk/component/internal/uuid"
 	ocsf "github.com/smithy-security/smithy/sdk/gen/com/github/ocsf/ocsf_schema/v1"
 )
 
-const errInvalidConstructorEmptyReason = "cannot be empty"
+const (
+	errInvalidConstructorEmptyReason = "cannot be empty"
+
+	columnNameFindings   columnName = "findings"
+	columnNameWorkflowID columnName = "workflow_id"
+	columnNameUpdatedAt  columnName = "updated_at"
+)
 
 type (
 	manager struct {
+		clock  clockwork.Clock
 		db     *sql.DB
 		dbName string
 	}
 
+	managerOption func(*manager) error
+
+	columnName string
+
 	// ErrInvalidConstructor should be used for invalid manager constructor errors.
 	ErrInvalidConstructor struct {
 		argName string
 		reason  string
 	}
 )
 
+// ManagerWithClock allows customising manager's clock.
+func ManagerWithClock(clock clockwork.Clock) managerOption {
+	return func(m *manager) error {
+		if clock == nil {
+			return errors.New("cannot set clock on nil clock")
+		}
+		m.clock = clock
+		return nil
+	}
+}
+
+func (cn columnName) String() string {
+	return string(cn)
+}
+
 func (e ErrInvalidConstructor) Error() string {
 	return fmt.Sprintf("invalid argument '%s': %s", e.argName, e.reason)
 }
 
 // NewManager returns a new SQLite database manager.
-func NewManager(dbPath string, dbName string) (*manager, error) {
-	switch {
-	case dbPath == "":
-		return nil, ErrInvalidConstructor{
-			argName: "db path",
-			reason:  errInvalidConstructorEmptyReason,
-		}
-	case dbName == "":
+func NewManager(dbPath string, dbName string, opts ...managerOption) (*manager, error) {
+	if dbName == "" {
```

Comment:
we should also check the `dbPath` over here that it's not empty and has a valid pattern